lib/core/src/chain_swap.rs

```diff
@@ -1218,9 +1264,25 @@ impl ChainSwapHandler {
         // Verify RBF
         let rbf_explicit = tx.input.iter().any(|tx_in| tx_in.sequence.is_rbf());
         if !verify_confirmation && rbf_explicit {
-            return Err(anyhow!("Transaction signals RBF"));
+            bail!("Transaction signals RBF");
         }
         // Verify amount
+        let actual_payer_amount_sat = match chain_swap.actual_payer_amount_sat {
+            Some(amount) => amount,
+            None => {
+                let actual_payer_amount_sat = self
+                    .fetch_incoming_swap_actual_payer_amount(chain_swap)
+                    .await?;
+                self.persister
+                    .update_actual_payer_amount(&chain_swap.id, actual_payer_amount_sat)?;
+                actual_payer_amount_sat
+            }
+        };
+        // For non-amountless swaps, make sure user locked up agreed amount
+        if chain_swap.payer_amount_sat > 0 && chain_swap.payer_amount_sat != actual_payer_amount_sat
+        {
+            bail!("Invalid server lockup tx - user lockup amount ({actual_payer_amount_sat} sat) differs from agreed ({} sat)", chain_swap.payer_amount_sat);
+        }
```

Comment:
I feel like for clarity this should be done in the `verify_user_lockup_tx()` fn, but I understand that it is only called for `TransactionServerConfirmed`, not `TransactionServerMempool`, so the swap state would already updated to Pending.

Reply:
Good point. Seems like we just forgot to add a call to verify_user_lockup_tx in that case? I don't see why it shouldn't be called in the mempool case.

Reply:
Thanks @dangeross! I've refactored and added the verification of the user lockup tx to the handling of TransactionServerMempool in [94ec703](https://github.com/breez/breez-sdk-liquid/pull/667/commits/94ec703414b6eb0eea54ed470290f58af1f6a753)